spring-kafka/src/main/java/org/springframework/kafka/listener/CompositeRecordInterceptor.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.kafka.listener;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+
+/**
+ * A {@link RecordInterceptor} that delegates to one or more {@link RecordInterceptor} in
+ * order.
+ *
+ * @author Artem Bilan
+ * @since 2.3
+ *
+ */
+public class CompositeRecordInterceptor<K, V> implements RecordInterceptor<K, V> {
+
+	private final Collection<RecordInterceptor<K, V>> delegates = new ArrayList<>();
+
+	@SafeVarargs
+	public CompositeRecordInterceptor(RecordInterceptor<K, V>... delegates) {
+		this.delegates.addAll(Arrays.asList(delegates));
+	}
+
+	@Override
+	public ConsumerRecord<K, V> intercept(ConsumerRecord<K, V> record) {
+		ConsumerRecord<K, V> recordToIntercept = record;
+		for (RecordInterceptor<K, V> delegate : delegates) {
+			recordToIntercept = delegate.intercept(recordToIntercept);
+		}
+		return recordToIntercept;
+	}
+
+}
```

Comment:
New like in the end of file

Reply:
:( I have a new STS and didn't configure